hr_contract_amendment/models/hr_contract.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+#   Copyright (C) 2018 EBII (https://www.saaslys.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, _
+
+
+class HrAmendmentContractChanges(models.Model):
+    _name = 'hr.contract.amendment.changes'
+    _description = 'HR contract amendment change'
+
+    amendment = fields.Many2one(
+        comodel_name='hr.contract.amendment', string="Amendment",
+        Require=True, ondelete="cascade")
+    value_name = fields.Char(string="Contract attribute name")
+    value_before = fields.Char(string="Value before change")
+    value_new = fields.Char(string="New value")
+
+
+class HrAmendmentContract(models.Model):
+    _name = 'hr.contract.amendment'
+    _description = 'HR contract amendment'
+
+    contract_id = fields.Many2one(
+        comodel_name='hr.contract', string="Referenced Contract",
+        required=True)
+    amendment_number = fields.Char(string="Amendment name")
+    date = fields.Date(required=True, default=fields.Date.context_today)
+    user_id = fields.Many2one('res.users', string='Registred by',
+                              default=lambda self: self.env.user)
+    amendment_type = fields.Selection(selection=[
+        ('JOB_CHANGE', _('JOB_CHANGE')),
+        ('SALARY_CHANGE', _('SALARY_CHANGE')),
+        ('WORKING_HOURS_CHANGE', _('WORKING_HOURS_CHANGE')),
+        ('CONTRACT_SITUATION', _('CONTRACT_SITUATION')),
+    ], string='Modification type')
+    modifications_ids = fields.One2many(
+        comodel_name="hr.contract.amendment.changes", inverse_name="amendment",
+        string="Changes", required=True, )
+
+
+class HrContract(models.Model):
+    _inherit = 'hr.contract'
+
+    amendment_ids = fields.One2many(
+        comodel_name='hr.contract.amendment', inverse_name="contract_id",
+        string="contract", required=False)
+
+    def write(self, vals):
+        sequence = self.env['ir.sequence'].next_by_code("amendment.number")
+        if 'job_id' in vals.keys() and len(vals) == 1:
```

Comment:
I would add some config to allow users to select fields for what they want to create an amendment, i don't want a new amendment for adding something is notes, advantages, change visa_expire...and probably to be linked with the type of change...

Reply:
sure test a list of fields
 how simple to do that ?